cmds/io/io_linux.go

```diff
@@ -8,18 +8,79 @@ package main
 import (
 	"encoding/binary"
 	"fmt"
-	"io"
+	"syscall"
+	"unsafe"
 )
 
-func in(f io.ReadSeeker, addr uint64, data interface{}) error {
+func in(f IoIntf, addr uint64, data interface{}) error {
+	ps := uint64(syscall.Getpagesize())
+	page := (addr & ^(ps - 1))
+	offset := addr - page
+	mem, err := syscall.Mmap(int(f.Fd()), int64(page), int(ps), syscall.PROT_READ, syscall.MAP_SHARED)
```

Comment:
I think you don't need to know pagesize for length, only base address. base addresses need to be page aligned, but not sizes.

Let's consider 32 bits accessed at 0xfff. You need to know the page size so you can get the alignment (which you do). 

So:
address := 0xfff
end := address + 4 // 32 bits
pageaddress := address & ^ps
length := end - pageaddress
mem, err := syscall.Mmap(int(f.Fd()), int64(pageaddress), length, syscall.PROT_READ, syscall.MAP_SHARED)

sizes in mmap don't have to be page aligned, just addresses. So this should do it.